### ISSUE TITLE
hvd tensorflow multi-gpu training hangs up after 190~220k steps

### ISSUE THREAD

**Issue by ewrfcas** [horovod/horovod#1078]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet) TensorFlow
2. Framework version: TensorFlow-gpu 1.13.1
3. Horovod version: 0.16.1
4. MPI version: openmpi 4.0.0
5. CUDA version: cuda10.0
6. NCCL version: v2.4.2 for cuda10
7. Python version: python 3.6
8. OS and version: ubuntu 16.04

I am using horovod to train a large model. The training can work properly as first. But after several steps(about 190k~220k steps), one process is died while other processes are hung for it (not sure). The usages of all gpus are 100%.
This picture shows the problem when the total processes are hung up. (pid 148832 to 148839, and the cpu% of 148833 is 0)
![25941557924876_ pic_hd](https://user-images.githubusercontent.com/12042259/57781040-f7861680-775b-11e9-8fab-6771187c1c17.jpg)
If i stop the training process and continue the training from the checkpoint, the training can work properly again. I have maintained the previous training situation contains the data generation with the same seed. But the same problem happened in 190k (start from 0), 290k (start from 100k), 420k (start from 200k) EVERY DAY.

I am looking for advices to find the possible cause for it.

**Comment by tgaddair** [comment]
Hey @ewrfcas, there is a known issue with TensorFlow 1.13.1 with Horovod (#876).  Can your script run against TensorFlow 1.12, and if so, could you try using that or `tf-nightly-gpu`?

**Comment by ewrfcas** [comment]
@tgaddair Thanks for your advice. But I don't have the problem in #876 issue. All gpus seem to be working well before one of them is hung up.
![image](https://user-images.githubusercontent.com/12042259/57825795-1673bf80-77d3-11e9-9d34-6319b9d06f30.png)

**Comment by alsrgv** [comment]
@ewrfcas, can you dump `thread apply all bt` using `gdb` from all processes when training gets stuck?

**Comment by ewrfcas** [comment]
@alsrgv , thanks. I will do it when the hang up happens.

**Comment by ewrfcas** [comment]
Hi @alsrgv, the process is hung up again in 611827 steps. 
The stuck process is pid=9065 and the message from gdb of it is
![image](https://user-images.githubusercontent.com/12042259/57855243-c6214f80-781c-11e9-8461-c6bad18be108.png)
The messages of other processes (9058, 9064...) are the same as it except Thread id and LWP id.

**Comment by romerojosh** [comment]
@ewrfcas If possible, can you try updating your NCCL library to a recent version (i.e. v2.4.7)? There was an issue with NCCL 2.4.2 where very long running jobs could hang that has been addressed in recent versions.

**Comment by ewrfcas** [comment]
@romerojosh thank you very much. I will try NCCL 2.4.7 to train the next model and give the feedbacks.

**Comment by ewrfcas** [comment]
@romerojosh thanks, and I solve this problem by reinsalling NCCL 2.4.7
